pkg/driver/controller_server.go

```diff
@@ -94,6 +94,20 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 			return nil, status.Errorf(codes.Unavailable, "Volume isn't available to be attached, state is currently %s", v.Status)
 		}
 	}
+
+	// TODO: Uncomment after client implementation is complete.
+	// snapshotID := ""
+	// if volSource := req.GetVolumeContentSource(); volSource != nil {
+	// 	if _, ok := volSource.GetType().(*csi.VolumeContentSource_Snapshot); !ok {
+	// 		return nil, status.Error(codes.InvalidArgument, "Unsupported volumeContentSource type")
+	// 	}
+	// 	snapshot := volSource.GetSnapshot()
+	// 	if snapshot == nil {
+	// 		return nil, status.Error(codes.InvalidArgument, "failed to get snapshot from the volumeContentSource")
+	// 	}
+	// 	snapshotID = snapshot.GetSnapshotId()
```

Comment:
I'm assuming this `snapshot.GetSnapshotId()` does not return an empty string -- but if it does, it would be very confusing from the code. Is this something we need to protect against, or should be safe to assume that the string input will not be empty?

Reply:
Thank you for your feedback!
Yes, according to [this spec](https://github.com/container-storage-interface/spec/blob/master/lib/go/csi/csi.pb.go#L6279), it is marked as required, so I implemented it under the assumption that it would not be empty. However, as you pointed out, adding a check for an empty string would be safer as a precaution. I will update the code to include this additional validation :+1:

Reply:
I have fixed it: https://github.com/civo/civo-csi/pull/39/commits/4dde3f12d77186451a73f16464c0881021b9d961